### DISCUSSION TITLE
Unabel to run evoMaster using mongoDB

### DISCUSSION THREAD

**Discussion by sai-teja2** [EMResearch/EvoMaster#1148] [Q&A]
I'm trying to connect EvoMaster with my code to perform white box testing.
here is my driver config 
`import com.aidtaas.mobius.hcy.web.MarketplaceApplication;
import com.mongodb.internal.connection.Connection;
import org.evomaster.client.java.controller.EmbeddedSutController;
import org.evomaster.client.java.controller.InstrumentedSutStarter;
import org.evomaster.client.java.controller.api.dto.SutInfoDto;
import org.evomaster.client.java.controller.api.dto.auth.AuthenticationDto;
import org.evomaster.client.java.controller.internal.SutController;
import org.evomaster.client.java.controller.problem.ProblemInfo;
import org.evomaster.client.java.controller.problem.RestProblem;
import org.evomaster.client.java.sql.DbSpecification;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.data.mongodb.core.MongoTemplate;

import java.util.List;

public class MyApplicationEvoMasterDriver extends EmbeddedSutController {

    public static void main(String[] args){

        SutController controller = new MyApplicationEvoMasterDriver();
        InstrumentedSutStarter starter = new InstrumentedSutStarter(controller);

        starter.start();
    }
    private ConfigurableApplicationContext ctx;
    private Connection connection;

    @Override
    public boolean isSutRunning() {
        return ctx.isRunning() && ctx!=null;
    }

    @Override
    public String getPackagePrefixesToCover() {
        return "com.aidtaas.mobius.hcy.web";
    }

    @Override
    public List<AuthenticationDto> getInfoForAuthentication() {
        return List.of();
    }

    @Override
    public ProblemInfo getProblemInfo() {

        return new RestProblem("http://localhost:9191/hcy-web/api/v3/api-docs", null);
    }

    @Override
    public SutInfoDto.OutputFormat getPreferredOutputFormat() {
        return SutInfoDto.OutputFormat.JAVA_JUNIT_5;
    }

    @Override
    public String startSut() {

        ctx = SpringApplication.run(MarketplaceApplication.class);

        MongoTemplate mongo = ctx.getBean(MongoTemplate.class);

        connection = (Connection) mongo.getCollection("holacracy");

        return "http://localhost:9191";

    }

    @Override
    public void stopSut() {
    ctx.stop();
    }

    @Override
    public void resetStateOfSUT() {

    }

    @Override
    public boolean handleLocalAuthenticationSetup(String authenticationInfo) {
        return super.handleLocalAuthenticationSetup(authenticationInfo);
    }

    @Override
    public List<DbSpecification> getDbSpecifications() {
        return List.of();
    }

}
`
The command I'm using to run Evo master

` java -jar /home/saiteja/Downloads/evomaster.jar --maxTime 20s --outputFolder /home/saiteja/Desktop/backend/hcy_20/hcy-web/src/test/java`

The error I have encountered 
`* 
 _____          ___  ___          _
|  ___|         |  \/  |         | |
| |____   _____ | .  . | __ _ ___| |_ ___ _ __
|  __\ \ / / _ \| |\/| |/ _` / __| __/ _ \ '__|
| |___\ V / (_) | |  | | (_| \__ \ ||  __/ |
\____/ \_/ \___/\_|  |_/\__,_|___/\__\___|_|

                    
* EvoMaster version: 3.0.0
* Loading configuration file from: /home/saiteja/em.yaml
* Initializing...
* [ERROR] ERROR: Cannot communicate with remote EvoMaster Driver for the system under test at localhost:40100
  Make sure the EvoMaster Driver for the system under test is running correctly.
`

FYI - My application runs on mongoDB, spring boot - 3.1.5, and java version 21

**Comment by arcuri82** [comment]
Hi,
 before running evomaster.jar, have you started the Driver process? 
if so, is it running on the same localhost?

**Comment by sai-teja2** [comment]
Hi,  @arcuri82,
I'm new to evoMaster can you please guide me on how to start the driver process?
Thank you.

**Comment by arcuri82** [comment]
https://www.youtube.com/watch?v=3mYxjgnhLEo

1:53

**Comment by sai-teja2** [comment]
Hi @arcuri82,

In the documentation of evoMaster it's mentioned as  
" NoSQL databases: (e.g., MongoDB) currently no support. It is a work in progress."
so as of now, evoMaster doesn't support mongoDB?

Thank you.

**Reply by arcuri82** [reply]
Hi @sai-teja2 ,

 ah, damn, this is embarrassing... it is actually supported since 3.3.0... but need to update documentation for it...

**Reply by arcuri82** [reply]
until we write the documentation, you can look at one of the example, eg:

https://github.com/WebFuzzing/EMB/blob/master/jdk_8_maven/em/embedded/rest/gestaohospital/src/main/java/em/embedded/br/com/codenation/hospital/EmbeddedEvoMasterController.java
